### PR TITLE
Improve references for PINRemoteImage and PINCache via Carthage

### PR DESCRIPTION
**Pull request by fumito-ito**
For Carthage, Build target must have references for dependencies as `Linked frameworks and Libraries`. To use Carthage and Cocoapods same time, I added a new build target for Carthage (AsyncDisplayKit-iOS).

This PR would resolve https://github.com/TextureGroup/Texture/issues/256 .

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/5048f88a5882b778a30a6a9d9448bdba/raw/7096970d7e6b0a1bec6e58432083476c30fd4cdd/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/39149e9e23dd352500704ba0946586bd/raw/7096970d7e6b0a1bec6e58432083476c30fd4cdd/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/d1d267d1d80a13074e8068505fb51c16/raw/d776bc07c844e2f779694ca8ec0e3a8ff6c098ee/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f5c574e8bb4890984f14017abf0254d5/raw/03bc88b0ff8a179258e63f696034af45d89ae159/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/9a7cdc86e73318c303e280dd62c39107/raw/42a12ebf8ff11510a15d4eed8cb8e92b88a1c194/log.txt)

File: AsyncDisplayKit.xcodeproj/xcshareddata/xcschemes/AsyncDisplayKit.xcscheme [hunk 1]
```diff
@@ -75,13 +84,6 @@
             ReferencedContainer = "container:AsyncDisplayKit.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
-      <EnvironmentVariables>
-         <EnvironmentVariable
-            key = "FB_REFERENCE_IMAGE_DIR"
-            value = "$(SOURCE_ROOT)/Tests/ReferenceImages"
-            isEnabled = "YES">
-         </EnvironmentVariable>
-      </EnvironmentVariables>
```
Comment (Adlai-Holler) [inline AsyncDisplayKit.xcodeproj/xcshareddata/xcschemes/AsyncDisplayKit.xcscheme]: Was it intentional to remove this and the `SkippedTests` chunk above? Let's put them back.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/8be7509d35317cb744d01fa345b33c4a/raw/e949aede2063f8ae2ff937c746c03d91f2a3e1e6/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/89ecf2c068bda5ce7a2fe00ec5d8083a/raw/79ec9770e453ab8a8a403e5f7a94940d8c5f07f9/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/44f4c8fc1b79c5f3061fc70e3bed5196/raw/7d6bc7359a91b2caff0f4be3a899d64bc6d7381a/log.txt)

**Comment by fumito-ito** [comment]
This pr has some problems. I close it until these are resolved.

**Pull request closed without merging**